pageserver/src/storage_sync/index.rs

```diff
@@ -350,24 +100,23 @@ pub struct IndexPart {
     /// Each of the layers present on remote storage.
     ///
     /// Additional metadata can might exist in `layer_metadata`.
-    timeline_layers: HashSet<RelativePath>,
+    pub timeline_layers: HashSet<RelativePath>,
 
-    /// Currently is not really used in pageserver,
-    /// present to manually keep track of the layer files that pageserver might never retrieve.
-    ///
-    /// Such "holes" might appear if any upload task was evicted on an error threshold:
-    /// the this layer will only be rescheduled for upload on pageserver restart.
+    /// FIXME: unused field. This should be removed, but that changes the on-disk format,
+    /// so we need to make sure we're backwards- (and maybe forwards-) compatible
```

Comment:
Quick glimpse at current `main` code suggests that we _could_ have timelines & owning pageservers out there that have some of those `missing_layers` files.
How do we deal with those?

Reply:
No, we've never used that field, it's always just been empty.

Reply:
I believe the deserialization code actually ignores any extra fields that are present in the json file, so I believe we could just remove the field altogether from this struct. But I didn't dare to do it as part of this massive PR, it seems like a change we could do separately.